build/configs/imxrt1020-evk/imxrt1020-evk_download.sh

```diff
@@ -149,8 +149,8 @@ echo "PARTIION NAMES: ${parts[@]}"
 
 if [[ "${CONFIG_APP_BINARY_SEPARATION}" == "y" ]]
 then
-	#Erase All Partitions
-	for (( i=0; i<$num; i++ ))
+	#Erase All Except Userfs Partitions
+	for (( i=0; i< $(( $num - 1 )); i++ ))
```

Comment:
How do you confirm that userfs is enabled always?

Reply:
This download script is for the current default defconfig.

Reply:
Module wise partitiion check is NOT part of this PR. That would address this.

Reply:
@btheosam I don't think so. The download script has to work with any configuration. User could disable userfs. It's up to user. We can't make user to enable it always.
If userfs is disabled, this code is definitely wrong.